website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+import openerp
+from openerp import http
+from openerp.http import request
+from openerp.osv import orm
+import json
+
+
+class Chd_website(http.Controller):
+
+    @http.route('/chd_init/', auth='public', website=True)
+    # need to append useless **kwargs to allow the openerp debug switch to work.
+    def start(self, selected_id = False, type = False, **kwargs):
+        partner = request.env.user.partner_id
+        if not partner:
+            return request.render(
+                'website_chd_product_configurator.no_partner', {})
+        product_template_model = request.env['product.template']
+        product_product_model = request.env['product.product']
+        pc_at_model = request.env[
+            'price.component.attribute.template']
+        product_type_model = request.env['product.type']
+        chd_finishing = request.env['product.finishing']
+        # when posting a selected product to configure
+        if selected_id:
+            curr_types = product_type_model.search([
+                ('product_option_ids', 'in', [selected_id])
+                ])
+            # explicitly cast selected id integer otherwise browse raises error
+            curr_product = product_template_model.browse([int(selected_id)])
+            curr_chd_price_component_ats = curr_product.attribute_template_ids
+            avail_accessories = product_product_model.browse(
+                curr_product.chd_accessoire_ids.ids)
+            return request.render(
+                'website_chd_product_configurator.configurator',
+                {
+                    'curr_product_id': curr_product,
+                    'curr_types': curr_types,
+                    'curr_chd_price_component_ats':
+                    curr_chd_price_component_ats,
+                    'avail_accessories': avail_accessories,
+                    })
+        """first iteration, loading the page"""
+        return request.render('website_chd_product_configurator.conf_start', {
+            'conf_products': product_template_model.search([(
+                'chd_origin_product', '=', True)
+                ]),
+            })
+
+    # shows the options list
+    @http.route('/chd_init/<int:id>/', website = True)
```

Comment:
make this `'/chd_init/<model("product.template"):curr_product_id>/'`, then the framework does the browse below for you and some sanity checking